stackable-certs/src/ca/mod.rs

```diff
@@ -294,13 +331,84 @@ where
         let key = ecdsa::SigningKey::new().context(GenerateEcdsaSigningKeySnafu)?;
         self.generate_leaf_certificate(key, name, scope, validity)
     }
+
+    /// Create a [`CertificateAuthority`] from a Kubernetes [`Secret`].
+    #[instrument(name = "create_certificate_authority_from_k8s_secret", skip(secret))]
+    pub fn from_secret(
+        secret: Secret,
+        key_certificate: &str,
+        key_private_key: &str,
+    ) -> Result<Self, SecretError<S::Error>> {
+        if !secret.type_.as_ref().is_some_and(|s| s == TLS_SECRET_TYPE) {
+            return InvalidSecretTypeSnafu.fail();
+        }
+
+        let data = secret.data.as_ref().with_context(|| NoSecretDataSnafu {
+            secret: ObjectRef::from_obj(&secret),
+        })?;
+
+        debug!("retrieving certificate data from secret via key {key_certificate:?}");
+        let certificate_data =
+            data.get(key_certificate)
+                .with_context(|| NoCertificateDataSnafu {
+                    secret: ObjectRef::from_obj(&secret),
+                })?;
+
+        let certificate = x509_cert::Certificate::load_pem_chain(&certificate_data.0)
+            .with_context(|_| ReadChainSnafu {
+                secret: ObjectRef::from_obj(&secret),
+            })?
+            .remove(0);
+
+        debug!("retrieving private key data from secret via key {key_certificate:?}");
+        let private_key_data =
+            data.get(key_private_key)
+                .with_context(|| NoPrivateKeyDataSnafu {
+                    secret: ObjectRef::from_obj(&secret),
+                })?;
+
+        let private_key_data =
+            std::str::from_utf8(&private_key_data.0).context(DecodeUtf8StringSnafu)?;
+
+        let signing_key_pair =
+            S::from_pkcs8_pem(private_key_data).context(DeserializeKeyFromPemSnafu)?;
+
+        Ok(Self {
+            certificate_pair: CertificatePair {
+                key_pair: signing_key_pair,
+                certificate,
+            },
+        })
+    }
+
+    /// Create a [`CertificateAuthority`] from a Kubernetes [`SecretReference`].
+    #[instrument(
+        name = "create_certificate_authority_from_k8s_secret_ref",
+        skip(secret_ref, client)
+    )]
+    pub async fn from_secret_ref(
+        secret_ref: &SecretReference,
+        key_certificate: &str,
+        key_private_key: &str,
```

Comment:
**nit (non-blocker)**: the duplicate `key` irks me a little.

```suggestion
        certificate: &str,
        private_key: &str,
```

Reply:
There is another instance further up in this file.

Reply:
Yeah this is a little weird, but we are talking about the _key_ of the key/value pair in the secret.

I don't know if there is any better way of naming these parameters.

Reply:
Oh, I didn't even consider it meant that (makes sense in the context), though I would maybe expect `key` to then be the suffix for both.

Eg:

```suggestion
        certificate_key: &str,
        private_key_key: &str,
```

or for more clarity, to [avoid ambiguity](https://docs.stackable.tech/home/nightly/contributor/code-style-guide#_long_versus_abbreviated)

```suggestion
        certificate_secret_key: &str,
        private_key_secret_key: &str,
```

Reply:
Added a doc comment in 4eb2743